SRC/NETWORK/hget.c

```diff
@@ -1409,9 +1487,49 @@ void CreateLocalFile()
 
 void UpdateReceivingMessage()
 {
+	static int sti=0;
+	unsigned char advance[4][3] = {"\r-","\r\\","\r|","\r/"};
+	static unsigned char ui = 0;
+
     if(!localFileIsConsole) {
-        PrintLongLength("\rDownloaded ", receivedLength, 1);
-        print(" ...");
+		if (contentLength)
+		{
+			if (currentBlock>=blockSize)
+			{
+				currentBlock-=blockSize;
+				print("=");
+			}
+
+			if (bFirstUpdate)
+			{
+				bFirstUpdate=false;
+				print("\r[                         ]");
+				__asm

```

Comment:
Please move this chunk of assembler to its own function with a descriptive name, this way the code will be easier to read.

Reply:
Done :)